src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -375,7 +381,15 @@ def upload_stream(
 
     def notify_query_listeners(self, query_record: QueryRecord) -> None:
         for listener in self._query_listener:
-            listener._add_query(query_record)
+            if listener.include_thread_id:
+                new_record = QueryRecord(
```

Comment:
you missed the async query listener. is that on purpose?

Reply:
did you mean`execute_async_and_notify_query_listener`? I think `execute_async_and_notify_query_listener` calls notify_query_listeners